astropy/coordinates/spectral_coordinate.py

```diff
@@ -0,0 +1,850 @@
+    return np.sqrt((1 + beta) / (1 - beta)) - 1
+
+
+def _redshift_to_velocity(redshift):
+    """
+    Convert a relativistic redshift to a velocity.
+    """
+    zponesq = (1 + redshift) ** 2
+    return c * (zponesq - 1) / (zponesq + 1)
+
+
+def update_differentials_to_match(original, velocity_reference, preserve_observer_frame=False):
+    """
+    Given an original coordinate object, update the differentials so that
+    the final coordinate is at the same location as the original coordinate
+    but co-moving with the velocity reference object.
+
+    If preserve_original_frame is set to True, the resulting object will be in
+    the frame of the original coordinate, otherwise it will be in the frame of
+    the velocity reference.
+    """
+
+    if not velocity_reference.data.differentials:
+        raise ValueError("Reference frame has no velocities")
+
+    # If the reference has an obstime already defined, we should ignore
+    # it and stick with the original observer obstime.
+    if 'obstime' in velocity_reference.frame_attributes and hasattr(original, 'obstime'):
+        velocity_reference = velocity_reference.replicate(obstime=original.obstime)
+
+    # We transform both coordinates to ICRS for simplicity and because we know
+    # it's a simple frame that is not time-dependent (it could be that both
+    # the original and velocity_reference frame are time-dependent)
+
+    original_icrs = original.transform_to(ICRS)
+    velocity_reference_icrs = velocity_reference.transform_to(ICRS)
+
+    differentials = velocity_reference_icrs.data.represent_as(CartesianRepresentation,
+                                                              CartesianDifferential).differentials
+
+    if original_icrs.data.differentials:
+        data_with_differentials = original_icrs.data.represent_as(CartesianRepresentation,
+                                                                  CartesianDifferential).with_differentials(differentials)
+    else:
+        data_with_differentials = original_icrs.data.represent_as(CartesianRepresentation).with_differentials(differentials)
+
+    final_icrs = original_icrs.realize_frame(data_with_differentials)
+
+    if preserve_observer_frame:
+        final = final_icrs.transform_to(original)
+    else:
+        final = final_icrs.transform_to(velocity_reference)
+
+    return final.replicate(representation_type=CartesianRepresentation,
+                           differential_type=CartesianDifferential)
+
+
+def attach_zero_velocities(coord):
+    """
+    Set the differentials to be stationary on a coordinate object.
+    """
+    coord_diffs = CartesianDifferential(u.Quantity([0, 0, 0] * u.km / u.s))
+    new_data = coord.data.to_cartesian().with_differentials(coord_diffs)
+    return coord.realize_frame(new_data)
+
+
+class SpectralCoord(u.SpectralQuantity):
+    """
+    Coordinate object representing spectral values.
+
+    The `SpectralCoord` class is new in Astropy v4.1 and should be considered
```

Comment:
should this be a `..note::` or similar rst formatted block?

Reply:
Yes, fixed!